tests/queries/test_query.py

```diff
@@ -165,6 +165,18 @@ def test_equality(self):
             Author.objects.filter(Q(item__name='foo')).query,
         )
 
+    def test_equality_with_negated_nullable(self):
+        # nullable field
+        self.assertEqual(
+            str(Item.objects.exclude(modified__year='2021').values('id').query),
+            str(Item.objects.annotate(year=ExtractYear('modified')).exclude(year='2021').values('id').query),
```

Comment:
`ExtractYear()` is not important for this test, I would use a simpler function, e.g. `Lower()`.